strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/services/impl/ArtifactEntryServiceTest.java

```diff
@@ -362,6 +364,80 @@ public void searchByTwoCoordinate(TestInfo testInfo)
         assertThat(c).isEqualTo(Long.valueOf(1));
     }
 
+    @Test
+    public void saveEntityCreationDateShouldBeGeneratedAutomaticallyAndRemainUnchanged(TestInfo testInfo)
+    {
+        final String groupId = getGroupId(GROUP_ID, testInfo);
+
+        final ArtifactEntry artifactEntry = createArtifactEntry(groupId);
+
+        assertThat(artifactEntry.getCreated()).isNull();
+        assertThat(artifactEntry.getLastUpdated()).isNull();
+
+        final ArtifactEntry firstTimeSavedArtifactEntry = save(artifactEntry);
+        final String artifactEntryId = firstTimeSavedArtifactEntry.getObjectId();
+        final Date creationDate = firstTimeSavedArtifactEntry.getCreated();
+
+        final ArtifactEntry firstTimeReadFromDatabase = artifactEntryService.findOne(artifactEntryId)
+                .orElse(null);
+
+        assertThat(firstTimeReadFromDatabase).isNotNull();
+        assertThat(firstTimeReadFromDatabase.getCreated()).isEqualTo(creationDate);
+
+        artifactEntry.setDownloadCount(1);
+        save(firstTimeReadFromDatabase);
+
+        final ArtifactEntry secondTimeReadFromDatabase = artifactEntryService.findOne(artifactEntryId)
+                .orElse(null);
+
+        assertThat(secondTimeReadFromDatabase).isNotNull();
+        assertThat(secondTimeReadFromDatabase.getCreated()).isEqualTo(creationDate);
+    }
+
+    @Test
+    public void saveEntityCreatedLastUsedLastUpdatedPropertiesShouldRetainTime(TestInfo testInfo) throws ParseException {
```

Comment:
Cosmetic: Braces and `throws` clause on new lines, please! Thanks! :)

Reply:
Done